server/views/pages/createGoals/index.njk

```diff
@@ -43,7 +43,7 @@
               {{ govukCharacterCount({
                 name: "title",
```

Comment:
All of the `name` and `id` fields need changing to reflect _which_ goal's / step's field it is
I think you want something like:
```
  name: "goals[0].title",
  id: "goals[0].title",
  value: form.goals[0].title,
```
And use the same pattern/approach for all the other fields